modules/data.remote/R/soilgrids.R

```diff
@@ -0,0 +1,224 @@
+  for (dep in seq_along(depths)) {
+    
+    # ---- could also replace using terra with just raster and extract similar to the landtrendr functions
+    # setup virtual raster URLs for each layer
+    soc_mean.url <- paste0(base_data_url, depths[dep],"_mean.vrt")
+    soc_Q0.05.url <- paste0(base_data_url, depths[dep], "_Q0.05.vrt")
+    soc_Q0.50.url <- paste0(base_data_url, depths[dep], "_Q0.5.vrt")
+    soc_Q0.95.url <- paste0(base_data_url, depths[dep], "_Q0.95.vrt")
+    
+    # create virtual rasters && extract SOC values - the original unit is dg/kg
+    soc_mean <- terra::extract(terra::rast(soc_mean.url), p_reproj)
+    soc_Q0.05_map <- terra::extract(terra::rast(soc_Q0.05.url), p_reproj)
+    soc_Q0.50_map <- terra::extract(terra::rast(soc_Q0.50.url), p_reproj)
+    soc_Q0.95_map <- terra::extract(terra::rast(soc_Q0.95.url), p_reproj)
+    # ---- could also replace using terra with just raster and extract similar to the landtrendr functions
+    
+    #change the unit to more common g/kg
+    soc_mean_real <- soc_mean[, 2] / 10
+    soc_Q0.05_real <- soc_Q0.05_map[, 2] / 10
+    soc_Q0.50_real <- soc_Q0.50_map[, 2] / 10
+    soc_Q0.95_real <- soc_Q0.95_map[, 2] / 10
+    
+    socquant[i,] <- c(soc_mean_real, soc_Q0.05_real, soc_Q0.50_real, soc_Q0.95_real)
+    siteids[i,] <- rep(as.numeric(internal_site_info$site_info.id), 4)
+    
+    rownames(socquant) <- c("0-5", "5-15", "15-30", "30-60", "60-100", "100-200")
+    colnames(socquant) <- c(rep("Mean", length(internal_site_info$site_info.lon)), 
+                            rep("0.05", length(internal_site_info$site_info.lon)),
+                            rep("0.5", length(internal_site_info$site_info.lon)),
+                            rep("0.95", length(internal_site_info$site_info.lon)))
+    rownames(siteids) <- c("0-5", "5-15", "15-30", "30-60", "60-100", "100-200")
+    colnames(siteids) <- c(rep("Mean", length(internal_site_info$site_info.lon)), 
+                           rep("0.05", length(internal_site_info$site_info.lon)), 
+                           rep("0.5", length(internal_site_info$site_info.lon)), 
+                           rep("0.95", length(internal_site_info$site_info.lon)))
+    i <- i + 1
+    
+    ### Display progress to console
+    if (verbose) {
+      setTxtProgressBar(pb, j)
+      j <- j+1
+      flush.console()
+    }
+    # cleanup interim results
+    rm(soc_mean.url, soc_Q0.05.url, soc_Q0.50.url, soc_Q0.95.url, 
+       soc_mean, soc_Q0.05_map, soc_Q0.50_map, soc_Q0.95_map,
+       soc_mean_real, soc_Q0.05_real, soc_Q0.50_real, soc_Q0.95_real)
+  }
+  
+  if (verbose) {
+    close(pb)
+  }
+  
+  # parse extracted data and prepare for output
+  soc_fit <- melt(socquant, id.vars = c("Mean"))
+  id_fit <- melt(siteids, id.vars = c("Mean"))
+  colnames(soc_fit) <- c("Depth", "Quantile", "Value")
+  soc_fit$Variable <- rep("soc", length(nrow(soc_fit)))
+  soc_fit$siteids <- id_fit$value
+  dat <- split(soc_fit, list(soc_fit$siteids, soc_fit$Depth))
```

Comment:
package

Reply:
updating to all being [package]::[function]

Reply:
done